docker/compose.yaml

```diff
@@ -1,63 +1,71 @@
+#  Copyright 2019 ABSA Group Limited
+#
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
 ---
-version: '3'
+name: 'spline-demo'
 services:
   arangodb:
     image: arangodb:${ARANGO_DB_VERSION}
     restart: on-failure
     ports:
       - ${ARANGO_DB_PORT}:8529
-    network_mode: "bridge"
     environment:
       ARANGO_NO_AUTH: 1
 
-  spline:
+  rest-server:
     image: absaoss/spline-rest-server:${SPLINE_CORE_VERSION}
     restart: on-failure
     ports:
       - ${SPLINE_REST_PORT}:8080
-    network_mode: "bridge"
     command: >
       bash -c "
-        until curl --output /dev/null --silent --get --fail http://172.17.0.1:${ARANGO_DB_PORT}/_db/spline/_admin/server/availability
+        until curl --output /dev/null --silent --get --fail http://arangodb:8529/_db/spline/_admin/server/availability
         do
-          echo waiting for 'spline' database to be ready
+          echo waiting for Spline database to be ready...
           sleep 5
         done
         catalina.sh run
       "
     environment:
-      SPLINE_DATABASE_CONNECTION_URL: 'arangodb://172.17.0.1:${ARANGO_DB_PORT}/spline'
+      SPLINE_DATABASE_CONNECTION_URL: 'arangodb://arangodb/spline'
       # by default /dev/random is used which may block
       CATALINA_OPTS: "-Dsecurerandom.source=file:/dev/./urandom -Djava.security.egd=file:/dev/./urandom"
-    links:
+    depends_on:
       - arangodb
```

Comment:
better solution could be:
```
  depends_on:
    db-init:
      condition: service_completed_successfully
```
logicaly `rest-server` depends on `db-init` which depends on `arangodb`

see similar solution here:
https://github.com/AbsaOSS/spline-openlineage/blob/master/docker/ol-integration-and-spline/docker-compose.yml

Reply:
Makes sense. Will update. Thank you.